### PR TITLE
Clarify exactly which scripts are witness outputs

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

**Comment by josibake** [comment]
ACK https://github.com/bitcoin/bips/pull/1537/commits/ff9feab538e7f814a694ad92f4487f998649b920

LGTM, clears up the confusion I had regarding `OP_PUSHBYTES_x` vs `OP_PUSHDATA1 x`.

File: bip-0141.mediawiki [hunk 1]
```diff
@@ -83,19 +83,23 @@ If all transactions in a block do not have witness data, the commitment is optio
 
 === Witness program ===
 
-A <code>scriptPubKey</code> (or <code>redeemScript</code> as defined in BIP16/P2SH) that consists of a 1-byte push opcode (for 0 to 16) followed by a data push between 2 and 40 bytes gets a new special meaning. The value of the first push is called the "version byte". The following byte vector pushed is called the "witness program".
+A <code>scriptPubKey</code> (or <code>redeemScript</code> as defined in BIP16/P2SH) that consists of a 1-byte push opcode (<code>OP_0</code> through <code>OP_16</code>) followed by a direct data push between 2 and 40 bytes gets a new special meaning. The value of the first push is called the "version byte". The following byte vector pushed is called the "witness program".
+In more detail, this means a <code>scriptPubKey</code> or <code>redeemScript</code> which consists of (in order):
+* First, byte 0x00 (<code>OP_0</code>) or any byte between 0x51 (<code>OP_1</code>) and 0x60 (<code>OP_16</code>) inclusive (the version byte).
+* Then, a byte ''L'' between 0x02 (push of 2 bytes) and 0x18 (push of 40 bytes) inclusive.
```
Comment (ajtowns) [inline bip-0141.mediawiki:89]: `0x18` should be `0x28` presumably

File: bip-0141.mediawiki [hunk 2]
```diff
@@ -83,19 +83,23 @@ If all transactions in a block do not have witness data, the commitment is optio
 
 === Witness program ===
 
-A <code>scriptPubKey</code> (or <code>redeemScript</code> as defined in BIP16/P2SH) that consists of a select subset of opcodes (<code>OP_0,OP_1,OP_2,...,OP_16</code>) followed by a data push between 2 and 40 bytes gets a new special meaning. The value of the first push is called the "version byte". The following byte vector pushed is called the "witness program".
+A <code>scriptPubKey</code> (or <code>redeemScript</code> as defined in BIP16/P2SH) that consists of a 1-byte push opcode (one of <code>OP_0,OP_1,OP_2,...,OP_16</code>) followed by a direct data push between 2 and 40 bytes gets a new special meaning. The value of the first push is called the "version byte". The following byte vector pushed is called the "witness program".
```
Comment (vostrnad) [inline bip-0141.mediawiki:86]: I'm not sure about the added word "direct" here. Perhaps "canonical" or "minimal" would be clearer?

Comment (vostrnad) [reply bip-0141.mediawiki:86]: Thank you for the clarification, the word "direct" makes complete sense now.

**Review by murchandamus — APPROVED** [review]
ACK 50e750a882f525cc9a297b75ca35682f3deb1f95

**Pull request merged**